chall02/jtrancos.py

```diff
@@ -0,0 +1,38 @@
+# **************************************************************************** #
+#                                                                              #
+#                                                         :::      ::::::::    #
+#    jtrancos.py                                        :+:      :+:    :+:    #
+#                                                     +:+ +:+         +:+      #
+#    By: jtrancos <marvin@42.fr>                    +#+  +:+       +#+         #
+#                                                 +#+#+#+#+#+   +#+            #
+#    Created: 2020/04/06 19:47:21 by jtrancos          #+#    #+#              #
+#    Updated: 2020/04/06 22:58:06 by jtrancos         ###   ########.fr        #
+#                                                                              #
+# **************************************************************************** #
+
+#!/usr/bin/env python
+
+import sys
+
+morse = { 
```

Comment:
Write your global names in upper case

Reply:
So something like this would be correct?


#!/usr/bin/env python3

import sys

Morse = {
        'A': '.-', 'a': '.-', 'B': '-...', 'b': '-...','C': '-.-.', 'c': '-.-.',        
        'D': '-..', 'd': '-..', 'E': '.', 'e': '.', 'F': '..-.', 'f': '..-.',
        'G': '--.', 'g': '--.', 'H': '....', 'h': '....', 'I': '..', 'i': '..',
        'J': '.---', 'j': '.---', 'K': '-.-', 'k': '-.-', 'L': '.-..', 'l': '.-..',     
        'M': '--', 'm': '--', 'N': '-.', 'n': '-.', 'O': '---', 'o': '---', 'P': '.--.',
        'p': '.--.', 'Q': '--.-', 'q': '--.-', 'R': '.-.', 'r': '.-.', 'S': '...',      
        's': '...', 'T': '-', 't': '-', 'U': '..-', 'u': '..-', 'V': '...-',
        'v': '...-', 'W': '.--', 'w': '.--', 'X': '-..-', 'x': '-..-', 'Y': '-.--',     
        'y': '-.--', 'Z': '--..', 'z': '--..', ' ': ' '
}

if __name__ == "__main__"
        if len(sys.argv) != 2 or sys.argv[1] == "":
                print ("usage: ./jtrancos.py <a-zA-Z string>")
                sys.exit()
        if all(x.isalpha() or x.isspace() for x in sys.argv[1]):
                text = ''
                for letter in sys.argv[1]:
                        text = text + (morse[letter])
                print(text)
        else:
                print ("usage: ./jtrancos.py <a-zA-Z string>")

Reply:
Well, that's a start 😄 
```python
#!/usr/bin/env python3

import sys

MORSE = {
'A': '.-',
'B': '-...', 
'C': '-.-.',
'D': '-..',
'E': '.',
'F': '..-.',
'G': '--.',
'H': '....', 
'I': '..',
'J': '.---',
'K': '-.-',
'L': '.-..',
'M': '--',
'N': '-.',
'O': '---', 
'P': '.--.',
'Q': '--.-',
'R': '.-.',
'S': '...',
'T': '-',
'U': '..-',
'V': '...-',
'W': '.--',
'X': '-..-',
'Y': '-.--',
'Z': '--..',
' ': ' '
}

def main():
  if len(sys.argv) != 2 or sys.argv[1] == "":
    print ("usage: ./jtrancos.py <a-zA-Z string>")
    sys.exit()
  if all(x.isalpha() or x.isspace() for x in sys.argv[1]):
    text = ''
    for letter in sys.argv[1]:
      text = text + (morse[letter])
    print(text)
  else:
    print ("usage: ./jtrancos.py  <a-zA-Z string>")

if __name__ == '__main__':
  main()
```

Reply:
You don't need the characters in upper case and in lower case. If you want to simplify your code, you can do a function outside for your error handling. Now that you don't have upper and lower case, instead of doing check with `isalpha` `isspace` you can look if your characters are in your `MORSE` variable doing `uppercase`. Don't forget too that your error message wasn't right, you forgot a part of it

Reply:
Oh nice I see, yeah that would make it easier, thank you! :)

And for the error message would something like this be better?:
`def error():
print "usage: sys.argv[0] <a-zA-Z string>"`

Reply:
Yes, for example. But with another function name because `error` could be use by your language, and be interpreted as other thing than your own function. 
Doing `print "usage: sys.argv[0] <a-zA-Z string>"` is a syntax of python2 in python3 you need to use `print("[...]")`
Don't forget your `exit` too.
It would be:
```python
def error():
  print(f"usage: {sys.argv[0]} <a-zA-Z string>")
  sys.exit()
```

Reply:
thank you!